test/autorest/lrogroup/lros_test.go

```diff
@@ -96,421 +96,532 @@ func TestLROBeginDelete202NoRetry204(t *testing.T) {
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp.RawResponse, 200)
-// }
+func TestLROBeginDeleteAsyncNoHeaderInRetry(t *testing.T) {
+	t.Skip("pending on CloudError field fix")
+	op := getLROSOperations(t)
+	resp, err := op.BeginDeleteAsyncNoHeaderInRetry(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller := resp.Poller
+	rt, err := poller.ResumeToken()
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller, err = op.ResumeDeleteAsyncNoHeaderInRetry(rt)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for !poller.Done() {
+		_, err := poller.Poll(context.Background())
+		if err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(200 * time.Millisecond)
+	}
+	resp, err = poller.FinalResponse(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	helpers.VerifyStatusCode(t, resp.RawResponse, 204)
+	resp1, err := op.BeginDeleteAsyncNoHeaderInRetry(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	resp, err = resp1.PollUntilDone(context.Background(), time.Duration(1)*time.Second)
+	if err != nil {
+		t.Fatal(err)
+	}
+	helpers.VerifyStatusCode(t, resp.RawResponse, 204)
+}
 
-// func TestLROBeginDeleteAsyncRetrySucceeded(t *testing.T) {
-// 	op := getLROSOperations(t)
-// 	poller, err := op.BeginDeleteAsyncRetrySucceeded(context.Background())
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	rt, err := poller.ResumeToken()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	poller, err = op.ResumeLrOSDeleteAsyncRetrySucceededPoller(rt)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	for poller.Poll(context.Background()) {
-// 		time.Sleep(200 * time.Millisecond)
-// 	}
-// 	resp, err := poller.Response()
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp.RawResponse, 200)
-// 	resp, err = poller.Wait(context.Background(), time.Duration(1)*time.Second)
-// 	if err != nil {
-// 		t.Fatal(err)
-// 	}
-// 	helpers.VerifyStatusCode(t, resp.RawResponse, 200)
-// }
+func TestLROBeginDeleteAsyncNoRetrySucceeded(t *testing.T) {
+	op := getLROSOperations(t)
+	resp, err := op.BeginDeleteAsyncNoRetrySucceeded(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller := resp.Poller
+	rt, err := poller.ResumeToken()
+	if err != nil {
+		t.Fatal(err)
+	}
+	poller, err = op.ResumeDeleteAsyncNoRetrySucceeded(rt)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for !poller.Done() {
+		_, err := poller.Poll(context.Background())
+		if err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(200 * time.Millisecond)
+	}
+	resp, err = poller.FinalResponse(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	helpers.VerifyStatusCode(t, resp.RawResponse, 202)
+	resp1, err := op.BeginDeleteAsyncNoRetrySucceeded(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	resp, err = resp1.PollUntilDone(context.Background(), time.Duration(1)*time.Second)
+	if err != nil {
+		t.Fatal(err)
+	}
+	helpers.VerifyStatusCode(t, resp.RawResponse, 202)
```

Comment:
This really ends with a 202?  Seems odd.

Reply:
The poller is not functioning correctly in this case :/ Looking into it

Reply:
Fixed